src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -553,16 +553,16 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
         if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
             $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
 
-            $legacyTranslationsDir = dirname($r->getFilename()).'/../../Resources/translations';
+            // with Symfony 2.4, the Security component was split into several subpackages
+            // and the translations have been moved to the symfony/security-core package
+            $translationsDir = dirname($r->getFilename()).'/../Resources/translations';
 
-            if (file_exists($legacyTranslationsDir)) {
+            if (!file_exists($translationsDir) && file_exists($dir = dirname($r->getFilename()).'/../../Resources/translations')) {
```

Comment:
Indeed, I did not notice that translations are contained in both directories (even if not both are complete). I guess the real issue was caused by allowing to install `symfony/security` and `symfony/security-core` in parallel (see symfony/symfony#16134) which led to the `AuthenticationException` being found first in the core package.

@nicolas-grekas Removing the translations from the core package means that we should use another class to detect the translations directory.